hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -589,11 +589,11 @@ object HoodieSparkSqlWriter {
     if (isCompatibleProjectionOf(tableSchema, newSchema)) {
       // Picking table schema as a writer schema we need to validate that we'd be able to
       // rewrite incoming batch's data (written in new schema) into it
-      (tableSchema, isSchemaCompatible(newSchema, tableSchema, true))
```

Comment:
Do you think we still need the validations inside `#deduceWriterSchema` ?

Reply:
I think we do. The check in `#deduceWriterSchema` is performed at creating rdd stage while `#validateSchema`  is part of writing stage. The first check allows us to identify schema incompatibility in advance